lib/httpClient.js

```diff
@@ -169,13 +175,40 @@ Client.prototype._destroyConnection = function () {
   this.conn.destroy()
 }
 
+Client.prototype._waitAndDestroy = function () {
+  const waitTimer = setTimeout(() => {
+    this._requestCompleted()
+  }, this.timeout)
+
+  this.on('response', () => {
+    clearTimeout(waitTimer)
+    this._requestCompleted()
+  })
+}
```

Comment:
Can we do this in `handleTimeout` instead? We could just avoid calling `this.timeoutTicker.clear()` during `destroy()` if there are ongoing requests.